### DISCUSSION TITLE
List of public RSS Box instances

### DISCUSSION THREAD

**Discussion by stefansundin** [stefansundin/rssbox#64] [Show and tell]
As the public Heroku instance is coming to an end (https://github.com/stefansundin/rssbox/issues/61), I wanted to create this discussion to serve as a directory of public RSS Box instances that I and other people host.

If you want to share your own public instance then please comment below. Thank you for helping the community!

If you want to donate to support RSS Box then the best option is to use [GitHub sponsors](https://github.com/sponsors/stefansundin). If that doesn't work then another easy option is to use [this PayPal link](https://www.paypal.com/donate/?cmd=_donations&business=KYMVEK692RF5Q&item_name=RSS+Box&no_shipping=1).

## Public instances provided by Stefan

**Note: These instances are quite slow so please do not overload them!**

- https://rssbox.fly.dev/
- https://rssbox.us-west-2.elasticbeanstalk.com/
  - This one might stop working at some point (probably in a year or two) since I don't like Elastic Beanstalk anymore. (I don't think I ever liked it actually)

## Public instances provided by others

_None so far_

**Comment by stefansundin** [comment]
If anyone started self-hosting RSS Box on fly.io then please make sure you are using a shared IPv4 address instead of a dedicated IPv4 address. They forgot to tell people they were going to charge for dedicated IPv4 addresses, and while NEW applications now get a shared IPv4 address by default, old applications retain their dedicated IPv4 addresses.

TL;DR: If you created your fly app around the same that I made the transition to fly myself, then you will need to run a few commands to migrate to a shared IPv4 address.

Here's their help article: https://fly.io/docs/reference/services/#shared-ipv4

I had to run these commands:

```shell
fly ips list
fly ips allocate-v4 --shared
fly ips release OLD_IPV4_ADDRESS_HERE
```

If you're impacted then you would probably have seen a surprise bill the other day, as they forgot to tell people about this change. They are refunding this, but I haven't seen anything saying that they will automatically migrate people to shared IPv4 addresses. So please check your app if you have one.

https://community.fly.io/t/we-are-going-to-start-charging-for-dedicated-ipv4-in-january-1st/15970/47

**Comment by stefansundin** [comment]
The Fly instance (https://rssbox.fly.dev/) may experience some slowness/throttling soon due to https://community.fly.io/t/predictable-processor-performance/22259/8. I don't know how bad it will be but it might be pretty bad, we'll see.

If you are having please try the Elastic Beanstalk instance (https://rssbox.us-west-2.elasticbeanstalk.com/).

I might have to start paying for the Fly instance, or figure out why it is using so much CPU.

**Reply by stefansundin** [reply]
I upgraded to Ruby 3.4 yesterday and this is looking much better now. I haven't really used the app in the last few months so I don't know how bad it was, but it was definitely throttled a lot. Here's a graph of the last 24 hours.

![Screenshot 2025-02-07 at 11 08 19](https://github.com/user-attachments/assets/efb45b39-0648-4c51-9b74-d5034b9e2955)

The red area means there's throttling happening. The baseline is the yellow dotted line at 6%.

**Reply by stefansundin** [reply]
Nevermind, after a few days it got worse again and started being throttled.

![Screenshot 2025-02-11 at 17 33 09](https://github.com/user-attachments/assets/f19612ec-ae0d-4b3f-8458-b1db47b30490)

Maybe a good solution is to restart the app every day, [similar to what is already done on Elastic Beanstalk](https://github.com/stefansundin/rssbox/blob/3646e857e2fda9ecb87dcd552496594589fbc40e/.ebextensions/30-restart.config).

**Reply by stefansundin** [reply]
I added a new way to restart the app every 24 hours in https://github.com/stefansundin/rssbox/commit/ff71cab8fe80bd00adbc491f544dbabe9fb8fe46, and deployed this 5 days ago and it appears to have been working great since then, no throttling so far.

<img width="1014" src="https://github.com/user-attachments/assets/98fb8ac1-46c5-48fe-b26f-f7e408e3c07f" />

**Reply by stefansundin** [reply]
For some reason it has been having issues again in the last few days.

I took a cursory look but couldn't pinpoint exactly what is causing it. I haven't deployed since [March 1](https://github.com/stefansundin/rssbox/releases/tag/fly%2Fv33), so I don't think it is anything I've done.

![Screenshot 2025-04-06 at 10 34 05](https://github.com/user-attachments/assets/a3f89fec-b332-4e27-8afa-dfb8ae7c51d3)

**Reply by stefansundin** [reply]
I solved the issue by running the app in "single mode" (by unsetting `WEB_CONCURRENCY`, in fda50f9ebf3cf0813338af03cf5174b4c50818b6), thus using less RAM and getting comfortably under the Fly limit for free apps.

<img width="1358" alt="Screenshot 2025-05-22 at 19 49 14" src="https://github.com/user-attachments/assets/10544de1-e995-4f89-bc98-2f40bbf86665" />

So this seems to be resolved for now. Sorry for the inconvenience.
